app/android/app_template/src/org/xwalk/app/template/AppTemplateActivity.java

```diff
@@ -35,24 +48,37 @@ protected void didTryLoadRuntimeView(View runtimeView) {
     }
 
     private void enterFullscreen() {
-        if (VERSION.SDK_INT >= VERSION_CODES.KITKAT &&
-                ((getWindow().getAttributes().flags &
-                        WindowManager.LayoutParams.FLAG_FULLSCREEN) != 0)) {
+        if (isNewerThanKitkat() && ((getWindow().getAttributes().flags &
+                WindowManager.LayoutParams.FLAG_FULLSCREEN) != 0)) {
             View decorView = getWindow().getDecorView();
-            decorView.setSystemUiVisibility(
-                    View.SYSTEM_UI_FLAG_LAYOUT_STABLE |
-                    View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION |
-                    View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN |
-                    View.SYSTEM_UI_FLAG_HIDE_NAVIGATION |
-                    View.SYSTEM_UI_FLAG_FULLSCREEN |
-                    View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY);
+            decorView.setSystemUiVisibility(SYSTEM_UI_OPTIONS);
         }
     }
 
     public void setIsFullscreen(boolean isFullscreen) {
         if (isFullscreen) {
             enterFullscreen();
+            if (isNewerThanKitkat()) {
+                setSystemUiVisibilityChangeListener();
+            }
         }
     }
 
+    private void setSystemUiVisibilityChangeListener() {
+        final View decorView = getWindow().getDecorView();
+        decorView.setOnSystemUiVisibilityChangeListener(
+            new View.OnSystemUiVisibilityChangeListener() {
+                @Override
+                public void onSystemUiVisibilityChange(int visibility) {
```

Comment:
Press volume button, ```onSystemUiVisibilityChange``` will be called, ```onResume()``` won't.
If the dial key was pressed, ```onSystemUiVisibilityChange``` will be called, presss back key to return, ```onSystemUiVisibilityChange``` will not be called again since the visibility was not changed, so call ```enterFullScreen``` in ```onResume```.